app/code/community/Drip/Connect/Helper/Data.php

```diff
@@ -119,19 +119,35 @@ public function isSubscriberExists($email, $storeId = null)
     /**
      * check if customer exists on the website
      *
+     * @param string $email
+     * @param Drip_Connect_Model_Configuration $config
+     *
      * @return bool
      */
-    public function isCustomerExists($email, $websiteId = null)
+    public function isCustomerExists($email, $config)
     {
-        if ($websiteId == null) {
-            $websiteId = Mage::app()->getStore()->getWebsiteId();
-        }
+        $websiteId = $config->getWebsiteId();
 
         $customer = Mage::getModel("customer/customer")->setWebsiteId($websiteId)->loadByEmail($email);
 
         return (bool) $customer->getId();
     }
 
+    /**
+     * find customer by email address
+     *
+     * @param string $email
+     * @param Drip_Connect_Model_Configuration $config
+     *
+     * @return Mage_Customer_Model_Customer
+     */
+    public function getCustomerByEmail($email, $config)
```

Comment:
Type hinting here as well:
```suggestion
    public function getCustomerByEmail($email, Drip_Connect_Model_Configuration $config)
```